pkg/cmd/pulumi/config_env.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/erikgeiser/promptkit/confirmation"
+	"github.com/pulumi/esc"
+	"github.com/pulumi/pulumi/pkg/v3/backend"
+	"github.com/pulumi/pulumi/pkg/v3/backend/display"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag/colors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	"github.com/spf13/cobra"
+)
+
+func newConfigEnvCmd(stack *string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "env",
+		Short: "Manage ESC environments for a stack",
+		Long: "Manages the ESC environment associated with a specific stack. To create a new environment\n" +
```

Comment:
This is all written very _stack_ specific. Do we not want ESC to play nicely with project config as well? I know we haven't got around to updating all the other config commands to deal with project config, but just want to check we're not painting ourselves into a corner vis-a-vis new ESC commands.

Reply:
> Do we not want ESC to play nicely with project config as well? 

We do. We don't have a specific design for that yet, but there are some ideas on https://github.com/pulumi/pulumi/issues/14499. I don't _think_ that we're painting ourselves into a corner there--AFAIK we don't have any commands for manipulating project config, but if we did I expect that they'd be implemented as flags to existing config commands.